File: app/src/main/java/com/channa/mobiledatausageapp/di/component/ApplicationComponent.java
package com.channa.mobiledatausageapp.di.component;

import android.content.Context;

import com.channa.mobiledatausageapp.di.module.ApplicationContextModule;
import com.channa.mobiledatausageapp.di.module.DataModule;
import com.channa.mobiledatausageapp.network.APIClient;
import com.channa.mobiledatausageapp.repository.MobileDataRepository;
import com.channa.mobiledatausageapp.view.MobileDataUsageActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationContextModule.class, DataModule.class})
public interface ApplicationComponent {

    void inject(MobileDataUsageActivity mobileDataUsageActivity);

    void inject(MobileDataRepository mobileDataRepository);

    APIClient apiClient();

    Context context();

}

File: app/src/main/java/com/zak/datauses/di/component/ApplicationComponent.kt
package com.zak.datauses.di.component

import android.content.Context
import com.zak.datauses.di.module.ApplicationContextModule
import com.zak.datauses.di.module.DataModule
import com.zak.datauses.network.APIClient
import com.zak.datauses.repository.MobileDataRepository
import com.zak.datauses.views.MobileDataUsageActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationContextModule::class, DataModule::class])
interface ApplicationComponent {
    fun inject(mobileDataUsageActivity: MobileDataUsageActivity?)
    fun inject(mobileDataRepository: MobileDataRepository?)
    fun apiClient(): APIClient?
    fun context(): Context?
}